File: app/src/main/java/me/zipi/navitotesla/api/NaverMapApi.java
package me.zipi.navitotesla.api;

import me.zipi.navitotesla.model.NaverMap;
import retrofit2.Call;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface NaverMapApi {


    @POST("/search2/searchMore.naver")
    @Headers({
            "Accept: application/json"
    })
    Call<NaverMap.Response> search(@Query("query") String query);

}

File: app/src/main/kotlin/me/zipi/navitotesla/api/NaverMapApi.kt
package me.zipi.navitotesla.api

import me.zipi.navitotesla.model.NaverMap
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface NaverMapApi {
    @GET("/v1/search/local.json")
    suspend fun search(
        @Query("query") query: String,
        @Query("display") display: Int = 5,
    ): Response<NaverMap.Response>
}
